src/workers.cc

```diff
@@ -848,6 +850,13 @@ void KafkaConsumerConsumeNumOfPartition::Execute() {
     // Get a message
     RdKafka::Message *message = queue->consume(timeout_ms);
     RdKafka::ErrorCode errorCode = message->err();
+
+    // Only wait for the first message. otherwise just fetch the current queue
+    // This allows getting ready messages, while not waiting for new ones
```

Comment:
What is a ready message and why are those interesting? I know a similar comment is present at the EOF error, but I would counter that we often use a timeout of 0, in which case a ready message would apparently also not be retrieved?

Reply:
Hmm that comment was indeed not really accurate. I've updated it.

Reply:
Check! And why 1 and not 0?

Reply:
i assumed it's the sensible thing to do, since the other timeout also is set to 1. I've dived a bit into librdkafka and this is what I found:

If the timeout is 0, a [blocking poll](https://github.com/confluentinc/librdkafka/blob/master/src/rdkafka.c#L3207-L3208) is done. The next line is a bit unclear to me: `rko = rd_kafka_q_pop(rkq, rd_timeout_remains_us(abs_timeout), 0)`. My assumption is that is returns an `rko` when there is a timeout, which can be used to poll the message via another thread. 

If that's the case, then a timeout of 1 will increase thread usage and be less blocking to the event loop. I think it makes sense to keep it.

Reply:
Okay! I reasoned that we in the past did `.setDefaultConsumeTimeout(0)` which is why I wanted to check.